osu.Game/Overlays/KeyBinding/KeyBindingRow.cs

```diff
@@ -408,8 +422,11 @@ public bool IsBinding
                 }
             }
 
-            public KeyButton(Framework.Input.Bindings.KeyBinding keyBinding)
+            public KeyButton(RealmKeyBinding keyBinding)
             {
+                if (keyBinding.IsManaged)
+                    throw new ArgumentException("Key binding should not be attached as we make temporary changes", nameof(keyBinding));
```

Comment:
Didn't quite realise this before I think, but the fact that a setting things on a model class may end up being a DB write worries me a touch. In this case I'd probably rather eagerly `.Detach()` the entity, but even that doesn't quite change the fact that now every time a realm model is passed around the user has to remember that it may be a live instance attached to DB and it needs to be double-checked before mutation.

I recall a `Live<T>` being proposed at some point, I think it would ease my concerns in this respect. Either that or go the other way around and add readonly structures/records.

Reply:
I think it was done this way to keep it simple and not have many abstractions over database model doing magic behind the scenes.

I would agree with immutability, though.

Reply:
Not willing to make any calls/changes on this until we get a better feel for realm. Because both of the concepts mentioned go against how realm is intended to be used (live data is basically the number one selling point in terms of getting its high performance, and as I found out in the previous implementation efforts, detaching or copying away can make it considerably slower than we would hope).